src/ui/proxy/interceptors.go

```diff
@@ -161,7 +161,7 @@ type readonlyHandler struct {
 func (rh readonlyHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	if config.ReadOnly() {
 		if req.Method == http.MethodDelete || req.Method == http.MethodPost || req.Method == http.MethodPatch {
-			http.Error(rw, "Upload/Delete is prohibited in read only mode.", http.StatusServiceUnavailable)
+			http.Error(rw, marshalError("DENIED", "Upload/Delete is prohibited in read only mode."), http.StatusForbidden)
```

Comment:
Can we extract this msg string "Upload/Delete is prohibited in read only mode." into a const ? Because it appears in two files in this commit.
And how about using  "The system is in read only mode. Any modification is not prohibited."

Reply:
The message could be refine.